runtime/src/main/java/io/quarkus/vault/runtime/VaultKvManager.java

```diff
@@ -8,50 +8,30 @@
 import jakarta.enterprise.inject.Produces;
 
 import io.quarkus.vault.VaultKVSecretReactiveEngine;
+import io.quarkus.vault.client.VaultClient;
+import io.quarkus.vault.client.api.secrets.kv1.VaultSecretsKV1;
+import io.quarkus.vault.client.api.secrets.kv2.VaultSecretsKV2;
+import io.quarkus.vault.client.api.secrets.kv2.VaultSecretsKV2ReadSecretData;
 import io.quarkus.vault.runtime.client.Private;
-import io.quarkus.vault.runtime.client.VaultClient;
-import io.quarkus.vault.runtime.client.dto.kv.VaultKvSecretV2WriteBody;
-import io.quarkus.vault.runtime.client.secretengine.VaultInternalKvV1SecretEngine;
-import io.quarkus.vault.runtime.client.secretengine.VaultInternalKvV2SecretEngine;
-import io.quarkus.vault.runtime.config.VaultRuntimeConfig;
 import io.smallrye.mutiny.Uni;
 
 @ApplicationScoped
 public class VaultKvManager implements VaultKVSecretReactiveEngine {
 
     @Produces
     @Private
-    public static VaultKvManager privateClientManager(@Private VaultClient vaultClient,
-            VaultAuthManager vaultAuthManager,
-            VaultConfigHolder vaultConfigHolder,
-            VaultInternalKvV1SecretEngine vaultInternalKvV1SecretEngine,
-            VaultInternalKvV2SecretEngine vaultInternalKvV2SecretEngine) {
-        return new VaultKvManager(vaultClient,
-                vaultAuthManager,
-                vaultConfigHolder,
-                vaultInternalKvV1SecretEngine,
-                vaultInternalKvV2SecretEngine);
+    public static VaultKvManager privateClientManager(@Private VaultClient vaultClient, VaultConfigHolder vaultConfigHolder) {
```

Comment:
do we still need this `Private` approach, when now you have the jdk and vertx client?
do we still have/need 2 vault clients (private and non private)?
can't we have just one managed vault client (ie: vertx), and for the config source we just create a simple jdk http client?

Reply:
The problem comes down to OpenTelemetry tracing. Using the JDK client we have no integration with OpenTelementry, so if config requests are made after initial application start (e.g. refreshes) they wouldn't be shown in traces.

I'm kind of Ok with them not being shown but this is how we did it before and it was very easy to replicate so I did.

Reply:
I would be ok dropping support opentelemetry for that particular use case.
I think this is one added value of what you did: providing a vault client that does not relate to quarkus.
we have gone through several issues regarding initialization because of the inter-dependency between quarkus and its configuration with regard to bootstrap.
if we make the config source to not rely on anything in quarkus, may be we will avoid future pitfalls.

Reply:
I'm in agreement. 

>we have gone through several issues regarding initialization because of the inter-dependency between quarkus and its configuration with regard to bootstrap.
if we make the config source to not rely on anything in quarkus, may be we will avoid future pitfalls.

This is exactly what I thought as well. This would remove the chicken/egg problem with the bootstrap entirely. If we're ok with losing OpenTelemetry for config requests, I'm onboard.

We might still need @Private and @Public it's just that @Private would be JDK. Unless we just manually build a JDK client in the config code (I'll need to look at it to see what's going on).

Reply:
@vsevel The JDK client doesn't have matching support for proxy configuration and the SSL configuration is complicated. It's possible but we'll need to write a custom `ProxySelector` to allow matching the Vert.x configuration.

I'm thinking, for now, we stick with what we have but get rid of @Shared, we'll just have @Private. Basically, we'll always use a @Private client (which has it's own dedicated Vertx instance) for the configuration lookups and then have an injectable client (with no qualifiers) that everything uses.

We can tackle switching to the simpler JDK in another PR. Thoughts?